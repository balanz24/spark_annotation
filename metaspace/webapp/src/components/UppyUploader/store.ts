// based on https://uppy.io/docs/stores/#Implementing-Stores

import { ref } from '@vue/composition-api'
import { State } from '@uppy/core'

type Listener = (prevState: State, nextState: State, patch: object) => void

export default () => {
  const state = ref<State>({})
  const listeners = new Set<Listener>()

  return {
    getState: () => state.value,
    setState: (patch: object) => {
      const prevState = state.value
      const nextState = { ...prevState, ...patch }

      state.value = nextState

      listeners.forEach(listener => {
        listener(prevState, nextState, patch)
      })
    },
    subscribe: (listener: Listener) => {
      listeners.add(listener)
      return () => listeners.delete(listener)
    },
  }
}
