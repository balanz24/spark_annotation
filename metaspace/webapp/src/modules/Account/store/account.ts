import { Module } from 'vuex'
import { DialogType } from '../dialogs'
import router from '../../../router'
import { RawLocation } from 'vue-router/types/router'

export interface AccountState {
  dialog: DialogType | null;
  dialogCloseRedirect: RawLocation | null;
  loginSuccessRedirect: RawLocation | null;
}

interface ShowDialogOptions {
  dialog: DialogType;
  dialogCloseRedirect?: RawLocation;
  loginSuccessRedirect?: RawLocation;
}

interface HideDialogOptions {
  dialog: DialogType;
  isLoginSuccess?: boolean;
}

const account: Module<AccountState, any> = {
  namespaced: true,
  state: {
    dialog: null,
    dialogCloseRedirect: null,
    loginSuccessRedirect: null,
  },
  mutations: {
    showDialog(state, payload: DialogType | ShowDialogOptions) {
      const options: ShowDialogOptions = typeof payload === 'string' ? { dialog: payload } : payload

      state.dialog = options.dialog
      if (options.dialogCloseRedirect != null) {
        state.dialogCloseRedirect = options.dialogCloseRedirect
      }
      if (options.loginSuccessRedirect != null) {
        state.loginSuccessRedirect = options.loginSuccessRedirect
      }
    },
    hideDialog(state, payload: DialogType | HideDialogOptions) {
      const options: HideDialogOptions = typeof payload === 'string' ? { dialog: payload } : payload

      if (state.dialog === options.dialog) {
        state.dialog = null
        if (options.isLoginSuccess && state.loginSuccessRedirect != null) {
          router.push(state.loginSuccessRedirect)
        } else if (state.dialogCloseRedirect != null) {
          router.push(state.dialogCloseRedirect)
        }

        state.dialogCloseRedirect = null
        state.loginSuccessRedirect = null
      }
    },
  },
}

export default account
