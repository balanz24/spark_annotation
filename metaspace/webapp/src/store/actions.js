import apolloClient from '../api/graphqlClient';
import {fetchOptionListsQuery} from '../api/metadata';
import {decodeParams} from '../modules/Filters';
import {computed} from "@vue/composition-api";
import store from "../store/index";
export default {

  async initFilterLists(context) {
    if(context.state.filterListsLoading || context.state.filterLists != null)
      return;

    context.commit('setFilterListsLoading');

    const response = await apolloClient.query({
      query: fetchOptionListsQuery,
      fetchPolicy: 'cache-first',
    });

    // set annotationIds default according to values passed from snapshot to the store state
    context.commit('setFilterLists', {...response.data,
      annotationIds: computed(() => store.state.snapshotAnnotationIds)} );

    // Refresh the current filter so that computed defaults that depend on `filterLists` are applied
    const filter = decodeParams(context.state.route, context.state.filterLists)
    context.commit('updateFilter', filter)
  },

};
