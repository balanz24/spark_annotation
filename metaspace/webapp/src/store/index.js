import Vue from 'vue';
import Vuex from 'vuex';
Vue.use(Vuex);

import getters from './getters.js';
import mutations from './mutations.js';
import actions from './actions.js';
import {accountModule} from '../modules/Account';

const store = new Vuex.Store({
  state: {
    // names of currently shown filters
    orderedActiveFilters: [],

    filterLists: null,
    filterListsLoading: false,

    // currently selected annotation
    annotation: undefined,

    // currently selected normalization matrix
    normalization: undefined,

    // roi settings
    roiInfo: {visible: false},

    // is annotation table loading?
    tableIsLoading: true,

    lastUsedFilters: {},

    currentTour: null,

    currentUser: {},

    // ion image global viewer settings
    channels: [],
    mode: 'SINGLE'
  },

  getters,
  mutations,
  actions,
  modules: {
    account: accountModule
  }
})

export default store;
