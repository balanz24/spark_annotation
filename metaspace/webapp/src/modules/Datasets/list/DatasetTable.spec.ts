import { mount } from '@vue/test-utils'
import DatasetTable from './DatasetTable.vue'
import router from '../../../router'
import { initMockGraphqlClient, apolloProvider } from '../../../../tests/utils/mockGraphqlClient'
import store from '../../../store/index'
import Vue from 'vue'
import Vuex from 'vuex'
import { sync } from 'vuex-router-sync'
import * as FileSaver from 'file-saver'
import { merge } from 'lodash-es'
import { mockGenerateId, resetGenerateId } from '../../../../tests/utils/mockGenerateId'
jest.mock('file-saver')
const mockFileSaver = FileSaver as jest.Mocked<typeof FileSaver>

Vue.use(Vuex)
sync(store, router)

const blobToText = (blob: Blob) => new Promise<string>((resolve, reject) => {
  const reader = new FileReader()
  reader.addEventListener('load', () => { resolve(reader.result as string) })
  reader.addEventListener('error', (e) => { reject((e as any).error) })
  reader.readAsText(blob)
})

describe('DatasetTable', () => {
  const mockMetadataJson = JSON.stringify({
    Metadata_Type: 'Imaging MS',
    MS_Analysis: {
      Detector_Resolving_Power: { mz: 1234, Resolving_Power: 123456 },
    },
  })
  const mockFdrCounts = {
    databaseId: 6,
    dbName: 'HMDB',
    dbVersion: 'v2.5',
    levels: [10],
    counts: [20],
  }
  const mockDataset = {
    id: 'REPLACEME',
    status: 'FINISHED',
    metadataJson: mockMetadataJson,
    description: null,
    databases: [
      { name: 'CHEBI', version: '', id: 2 },
      { name: 'HMDB', version: 'v2.5', id: 6 },
      { name: 'HMDB', version: 'v4', id: 22 },
    ],
    polarity: 'POSITIVE',
    fdrCounts: mockFdrCounts,
    groupApproved: true,
    canEdit: false,
    canDelete: false,
  }

  afterEach(() => {
    jest.useRealTimers()
    resetGenerateId()
  })

  it('should match snapshot', async() => {
    mockGenerateId(123)
    initMockGraphqlClient({
      Query: () => ({
        allDatasets: () => {
          return [
            { ...mockDataset, id: 'ANNOTATING1', status: 'ANNOTATING' },
            { ...mockDataset, id: 'QUEUED1', status: 'QUEUED' },
            { ...mockDataset, id: 'FINISHED1', status: 'FINISHED' },
          ]
        },
        countDatasetsPerGroup: () => ({
          counts: [
            { fieldValues: ['QUEUED'], count: 2 },
            { fieldValues: ['ANNOTATING'], count: 1 },
            { fieldValues: ['FINISHED'], count: 20 },
          ],
        }),
      }),
    })
    const wrapper = mount(DatasetTable, { parentComponent: { store, router }, apolloProvider })
    await Vue.nextTick()

    expect(wrapper.element).toMatchSnapshot()
  })

  it('should be able to export a CSV', async() => {
    initMockGraphqlClient({
      Query: () => ({
        allDatasets: (_: any, { filter: { status }, offset }: any) => {
          return [
            merge({}, mockDataset, { principalInvestigator: null, id: `${status}1`, status }),
            merge({}, mockDataset, { principalInvestigator: null, groupApproved: false, id: `${status}2`, status }),
            merge({}, mockDataset, {
              principalInvestigator: null,
              group: { adminNames: ['group', 'admin', 'names'] },
              id: `${status}3`,
              status,
            }),
            merge({}, mockDataset, {
              principalInvestigator: { name: 'principal investigator' },
              id: `${status}4`,
              status,
            }),
          ].slice(offset, offset + 2)
        },
        countDatasets: () => 4,
      }),
    })
    const wrapper = mount(DatasetTable, { parentComponent: { store, router }, apolloProvider })
    wrapper.setData({ csvChunkSize: 2 })
    await Vue.nextTick()

    await (wrapper.vm as any).startExport()
    expect(mockFileSaver.saveAs).toBeCalled()
    const blob: Blob = mockFileSaver.saveAs.mock.calls[0][0]
    const csv = await blobToText(blob)
    const csvWithoutDateHeader = csv.replace(/# Generated at .*\n/, '')

    expect(csvWithoutDateHeader).toMatchSnapshot()
  })
})
