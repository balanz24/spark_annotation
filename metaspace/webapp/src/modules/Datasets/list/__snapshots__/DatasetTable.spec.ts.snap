// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetTable should be able to export a CSV 1`] = `
"# URL: http://localhost/?ds_owner
\\"datasetId\\",\\"datasetName\\",\\"group\\",\\"submitter\\",\\"PI\\",\\"organism\\",\\"organismPart\\",\\"condition\\",\\"growthConditions\\",\\"ionisationSource\\",\\"maldiMatrix\\",\\"analyzer\\",\\"resPower400\\",\\"polarity\\",\\"uploadDateTime\\",\\"FDR@10%\\",\\"database\\",\\"opticalImage\\"
\\"FINISHED1\\",\\"datasets.0.name\\",\\"datasets.0.group.shortName\\",\\"datasets.0.submitter.name\\",\\"datasets.0.group.adminNames, datasets.0.group.adminNames\\",\\"datasets.0.organism\\",\\"datasets.0.organismPart\\",\\"datasets.0.condition\\",\\"datasets.0.growthConditions\\",\\"datasets.0.ionisationSource\\",\\"datasets.0.maldiMatrix\\",\\"datasets.0.analyzer.type\\",\\"13\\",\\"positive\\",\\"datasets.0.uploadDateTime\\",\\"20\\",\\"HMDB - v2.5\\",\\"http://localhostdatasets.0.rawOpticalImageUrl\\"
\\"FINISHED2\\",\\"datasets.1.name\\",\\"datasets.1.group.shortName\\",\\"datasets.1.submitter.name\\",\\"datasets.1.group.adminNames, datasets.1.group.adminNames\\",\\"datasets.1.organism\\",\\"datasets.1.organismPart\\",\\"datasets.1.condition\\",\\"datasets.1.growthConditions\\",\\"datasets.1.ionisationSource\\",\\"datasets.1.maldiMatrix\\",\\"datasets.1.analyzer.type\\",\\"13\\",\\"positive\\",\\"datasets.1.uploadDateTime\\",\\"20\\",\\"HMDB - v2.5\\",\\"http://localhostdatasets.1.rawOpticalImageUrl\\"
\\"FINISHED3\\",\\"datasets.0.name\\",\\"datasets.0.group.shortName\\",\\"datasets.0.submitter.name\\",\\"group, admin, names\\",\\"datasets.0.organism\\",\\"datasets.0.organismPart\\",\\"datasets.0.condition\\",\\"datasets.0.growthConditions\\",\\"datasets.0.ionisationSource\\",\\"datasets.0.maldiMatrix\\",\\"datasets.0.analyzer.type\\",\\"13\\",\\"positive\\",\\"datasets.0.uploadDateTime\\",\\"20\\",\\"HMDB - v2.5\\",\\"http://localhostdatasets.0.rawOpticalImageUrl\\"
\\"FINISHED4\\",\\"datasets.1.name\\",\\"datasets.1.group.shortName\\",\\"datasets.1.submitter.name\\",\\"principal investigator\\",\\"datasets.1.organism\\",\\"datasets.1.organismPart\\",\\"datasets.1.condition\\",\\"datasets.1.growthConditions\\",\\"datasets.1.ionisationSource\\",\\"datasets.1.maldiMatrix\\",\\"datasets.1.analyzer.type\\",\\"13\\",\\"positive\\",\\"datasets.1.uploadDateTime\\",\\"20\\",\\"HMDB - v2.5\\",\\"http://localhostdatasets.1.rawOpticalImageUrl\\"
"
`;

exports[`DatasetTable should match snapshot 1`] = `
<div>
  <div
    class="filter-panel"
  >
    <mock-el-select
      class="filter-select"
      placeholder="Add filter"
    >
      <mock-el-option
        label="Select group"
        value="group"
      />
      <mock-el-option
        label="Select project"
        value="project"
      />
      <mock-el-option
        label="Select submitter"
        value="submitter"
      />
      <mock-el-option
        label="Select dataset"
        value="datasetIds"
      />
      <mock-el-option
        label="Search molecule"
        value="compoundName"
      />
      <mock-el-option
        label="Select polarity"
        value="polarity"
      />
      <mock-el-option
        label="Select organism"
        value="organism"
      />
      <mock-el-option
        label="Select organism part"
        value="organismPart"
      />
      <mock-el-option
        label="Select organism condition"
        value="condition"
      />
      <mock-el-option
        label="Select sample growth conditions"
        value="growthConditions"
      />
      <mock-el-option
        label="Select analyzer"
        value="analyzerType"
      />
      <mock-el-option
        label="Select ionisation source"
        value="ionisationSource"
      />
      <mock-el-option
        label="Select MALDI matrix"
        value="maldiMatrix"
      />
      <mock-el-option
        label="Search anything"
        value="simpleQuery"
      />
      <mock-el-option
        label="Select data type"
        value="datasetOwner"
      />
      <mock-el-option
        label="Select data type"
        value="metadataType"
      />
      <mock-el-option
        label="Show/hide datasets with optical images"
        value="opticalImage"
      />
    </mock-el-select>
     
  </div>
   
  <div>
    <form
      class="el-form el-form--inline"
      id="dataset-list-header"
    >
      <div
        class="el-radio-group p-2 flex flex-row items-center"
        role="radiogroup"
      >
        <label
          aria-checked="true"
          class="el-radio-button el-radio-button--small is-active"
          role="radio"
          tabindex="0"
        >
          <input
            class="el-radio-button__orig-radio"
            tabindex="-1"
            type="radio"
            value="List"
          />
          <span
            class="el-radio-button__inner"
          >
            List
          </span>
        </label>
         
        <label
          class="el-radio-button el-radio-button--small"
          role="radio"
          tabindex="-1"
        >
          <input
            class="el-radio-button__orig-radio"
            tabindex="-1"
            type="radio"
            value="Summary"
          />
          <span
            class="el-radio-button__inner"
          >
            Summary
          </span>
        </label>
      </div>
       
      <div
        aria-label="checkbox-group"
        class="el-checkbox-group p-2 dataset-status-checkboxes"
        mock-v-loading="0"
        role="group"
      >
        <label
          class="el-checkbox cb-annotating is-checked"
        >
          <span
            class="el-checkbox__input is-checked"
          >
            <span
              class="el-checkbox__inner"
            />
            <input
              aria-hidden="false"
              class="el-checkbox__original"
              type="checkbox"
              value="ANNOTATING"
            />
          </span>
          <span
            class="el-checkbox__label"
          >
            
          Processing (1)
        
            <!---->
          </span>
        </label>
         
        <label
          class="el-checkbox cb-queued is-checked"
        >
          <span
            class="el-checkbox__input is-checked"
          >
            <span
              class="el-checkbox__inner"
            />
            <input
              aria-hidden="false"
              class="el-checkbox__original"
              type="checkbox"
              value="QUEUED"
            />
          </span>
          <span
            class="el-checkbox__label"
          >
            
          Queued (2)
        
            <!---->
          </span>
        </label>
         
        <label
          class="el-checkbox is-checked"
        >
          <span
            class="el-checkbox__input is-checked"
          >
            <span
              class="el-checkbox__inner"
            />
            <input
              aria-hidden="false"
              class="el-checkbox__original"
              type="checkbox"
              value="FINISHED"
            />
          </span>
          <span
            class="el-checkbox__label"
          >
            
          Finished (20)
        
            <!---->
          </span>
        </label>
         
        <!---->
      </div>
       
      <div
        style="flex-grow: 1;"
      />
       
      <div
        class="flex flex-row items-center justify-between flex-1"
      >
        <div
          class="p-2 flex flex-row sort-dp-container"
        >
          <mock-el-select
            clearable="true"
            placeholder="Sort by"
            size="small"
            value=""
          >
            <mock-el-option
              label="Last updated"
              value="ORDER_BY_DATE"
            />
            <mock-el-option
              label="Upload date"
              value="ORDER_BY_UP_DATE"
            />
            <mock-el-option
              label="Dataset name"
              value="ORDER_BY_NAME"
            />
            <mock-el-option
              label="Submitter name"
              value="ORDER_BY_DS_SUBMITTER_NAME"
            />
            <mock-el-option
              label="Annotation count"
              value="ORDER_BY_ANNOTATION_COUNTS"
            />
          </mock-el-select>
          <div
            class="el-input-group__append sort-dp-btn"
          >
            <mock-el-tooltip
              content="Sorting order"
              placement="right"
            >
              <button
                class="el-button el-button--default cursor-not-allowed"
                type="button"
              >
                <!---->
                <i
                  class="el-icon-sort"
                />
                <!---->
              </button>
            </mock-el-tooltip>
          </div>
        </div>
         
        <button
          class="el-button el-button--default el-button--small"
          type="button"
        >
          <!---->
          <!---->
          <span>
            
          Export to CSV
        
          </span>
        </button>
      </div>
    </form>
  </div>
   
  <div
    class="dataset-list allow-double-column"
    mock-v-loading="0"
  >
    <div
      class="dataset-item border border-solid border-gray-200 leading-5"
    >
      <div
        class="opt-image--container"
      >
        <div
          class="thumb"
        >
          <img
            alt="Optical image"
            class="thumb--img"
            src="allDatasets.1.thumbnailOpticalImageUrl"
            style="transform: scaleY(1) translateY(0%); height: 100px;"
          />
           
          <img
            alt="Ion image thumbnail"
            class="thumb--img thumb--img__ion"
            src="allDatasets.1.ionThumbnailUrl"
            style="transform: scaleY(1) translateY(0%); height: 100px;"
          />
           
          <!---->
        </div>
      </div>
       
      <div
        class="ds-info"
      >
        <div
          class="ds-item-line flex"
        >
          <span
            class="font-bold truncate"
            title="allDatasets.1.name"
          >
            allDatasets.1.name
          </span>
          <mock-el-popover
            class="ml-1"
            placement="top"
            trigger="hover"
          >
            <div
              slot-key="default"
            >
              <div
                mock-v-loading="true"
              />
            </div>
            <div
              slot-key="reference"
            >
              <i
                class="el-icon-lock"
              />
            </div>
          </mock-el-popover>
          <mock-el-popover
            class="ml-1"
            placement="top"
            popper-class="min-w-0"
            trigger="hover"
          >
            <div
              slot-key="default"
            >
               
              <p
                class="leading-none m-0"
              >
                Copy dataset id to clipboard
              </p>
            </div>
            <div
              slot-key="reference"
            >
              <button
                class="button-reset flex  w-6 h-6"
              >
                <svg
                  class="h-full w-full text-gray-400 hover:text-gray-500 sm-copy-icon"
                  data-icon="copy-id"
                />
              </button>
            </div>
          </mock-el-popover>
        </div>
        <div
          class="ds-item-line text-gray-700"
        >
          <span
            class="ds-add-filter"
            title="Filter by species"
          >
            allDatasets.1.organism
          </span>
          , 
          <span
            class="ds-add-filter"
            title="Filter by organism part"
          >
            alldatasets.1.organismpart
          </span>
           
          <span
            class="ds-add-filter"
            title="Filter by condition"
          >
            (alldatasets.1.condition)
          </span>
        </div>
        <div
          class="ds-item-line"
        >
          <span
            class="ds-add-filter"
            title="Filter by ionisation source"
          >
            allDatasets.1.ionisationSource
          </span>
           + 
          <span
            class="ds-add-filter"
            title="Filter by analyzer type"
          >
            allDatasets.1.analyzer.type
          </span>
          , 
          <span
            class="ds-add-filter"
            title="Filter by polarity"
          >
            positive mode
          </span>
          , RP 123k @ 1234
        </div>
        <div
          class="ds-item-line"
        >
          Submitted 
          <mock-elapsed-time
            date="allDatasets.1.uploadDT"
          />
           by 
          <span
            class="ds-add-filter"
            title="Filter by submitter"
          >
            allDatasets.1.submitter.name
          </span>
          <span>
            , 
            <mock-el-dropdown
              placement="bottom"
              show-timeout="50"
              trigger="hover"
            >
              <div
                slot-key="default"
              >
                <span
                  class="text-base text-primary cursor-pointer"
                >
                  allDatasets.1.group.shortName
                </span>
              </div>
              <div
                slot-key="dropdown"
              >
                <mock-el-dropdown-menu>
                  <mock-el-dropdown-item
                    command="filter_group"
                  >
                    Filter by this group
                  </mock-el-dropdown-item>
                  <mock-el-dropdown-item
                    command="view_group"
                  >
                    View group
                  </mock-el-dropdown-item>
                </mock-el-dropdown-menu>
              </div>
            </mock-el-dropdown>
          </span>
        </div>
      </div>
       
      <div
        class="ds-actions"
      >
        <mock-el-dialog
          customclass="dataset-item-dialog"
          title="Provided metadata"
        >
          <div
            class="el-row"
          >
            <mock-el-tree
              data="[object Object],[object Object],[object Object]"
              default-expanded-keys="MS analysis,Detector resolving power,Data Management,Annotation settings"
              id="metadata-tree"
              node-key="id"
            />
          </div>
        </mock-el-dialog>
        <span>
          <div
            class="striped-progressbar queued"
            title="Waiting in the queue"
          />
        </span>
        <div>
          <i
            class="el-icon-view"
          />
          <a
            href="#"
          >
            Show full metadata
          </a>
        </div>
        <div
          class="featured-action"
        >
          <i
            class="el-icon-data-analysis"
          />
          <div
            class="el-badge sm-feature-badge"
          >
            <a
              class="mr-2"
              href="/dataset/QUEUED1"
            >
              <span>
                Dataset overview
              </span>
            </a>
            <transition-stub
              name="el-zoom-in-center"
            >
              <sup
                class="el-badge__content el-badge__content--undefined is-fixed"
              >
                New
              </sup>
            </transition-stub>
          </div>
        </div>
      </div>
    </div>
    <div
      class="dataset-item border border-solid border-gray-200 leading-5"
    >
      <div
        class="opt-image--container"
      >
        <div
          class="thumb"
        >
          <img
            alt="Optical image"
            class="thumb--img"
            src="allDatasets.0.thumbnailOpticalImageUrl"
            style="transform: scaleY(1) translateY(0%); height: 100px;"
          />
           
          <img
            alt="Ion image thumbnail"
            class="thumb--img thumb--img__ion"
            src="allDatasets.0.ionThumbnailUrl"
            style="transform: scaleY(1) translateY(0%); height: 100px;"
          />
           
          <!---->
        </div>
      </div>
       
      <div
        class="ds-info"
      >
        <div
          class="ds-item-line flex"
        >
          <span
            class="font-bold truncate"
            title="allDatasets.0.name"
          >
            allDatasets.0.name
          </span>
          <mock-el-popover
            class="ml-1"
            placement="top"
            popper-class="min-w-0"
            trigger="hover"
          >
            <div
              slot-key="default"
            >
               
              <p
                class="leading-none m-0"
              >
                Copy dataset id to clipboard
              </p>
            </div>
            <div
              slot-key="reference"
            >
              <button
                class="button-reset flex  w-6 h-6"
              >
                <svg
                  class="h-full w-full text-gray-400 hover:text-gray-500 sm-copy-icon"
                  data-icon="copy-id"
                />
              </button>
            </div>
          </mock-el-popover>
        </div>
        <div
          class="ds-item-line text-gray-700"
        >
          <span
            class="ds-add-filter"
            title="Filter by species"
          >
            allDatasets.0.organism
          </span>
          , 
          <span
            class="ds-add-filter"
            title="Filter by organism part"
          >
            alldatasets.0.organismpart
          </span>
           
          <span
            class="ds-add-filter"
            title="Filter by condition"
          >
            (alldatasets.0.condition)
          </span>
        </div>
        <div
          class="ds-item-line"
        >
          <span
            class="ds-add-filter"
            title="Filter by ionisation source"
          >
            allDatasets.0.ionisationSource
          </span>
           + 
          <span
            class="ds-add-filter"
            title="Filter by analyzer type"
          >
            allDatasets.0.analyzer.type
          </span>
          , 
          <span
            class="ds-add-filter"
            title="Filter by polarity"
          >
            positive mode
          </span>
          , RP 123k @ 1234
        </div>
        <div
          class="ds-item-line"
        >
          Submitted 
          <mock-elapsed-time
            date="allDatasets.0.uploadDT"
          />
           by 
          <span
            class="ds-add-filter"
            title="Filter by submitter"
          >
            allDatasets.0.submitter.name
          </span>
          <span>
            , 
            <mock-el-dropdown
              placement="bottom"
              show-timeout="50"
              trigger="hover"
            >
              <div
                slot-key="default"
              >
                <span
                  class="text-base text-primary cursor-pointer"
                >
                  allDatasets.0.group.shortName
                </span>
              </div>
              <div
                slot-key="dropdown"
              >
                <mock-el-dropdown-menu>
                  <mock-el-dropdown-item
                    command="filter_group"
                  >
                    Filter by this group
                  </mock-el-dropdown-item>
                  <mock-el-dropdown-item
                    command="view_group"
                  >
                    View group
                  </mock-el-dropdown-item>
                </mock-el-dropdown-menu>
              </div>
            </mock-el-dropdown>
          </span>
        </div>
      </div>
       
      <div
        class="ds-actions"
      >
        <mock-el-dialog
          customclass="dataset-item-dialog"
          title="Provided metadata"
        >
          <div
            class="el-row"
          >
            <mock-el-tree
              data="[object Object],[object Object],[object Object]"
              default-expanded-keys="MS analysis,Detector resolving power,Data Management,Annotation settings"
              id="metadata-tree"
              node-key="id"
            />
          </div>
        </mock-el-dialog>
        <span>
          <div
            class="striped-progressbar processing"
            title="Processing is under way"
          />
        </span>
        <div>
          <i
            class="el-icon-view"
          />
          <a
            href="#"
          >
            Show full metadata
          </a>
        </div>
        <div>
          <i
            class="el-icon-data-analysis"
          />
          <a
            class=""
            classname="mr-2"
            href="/dataset/ANNOTATING1"
          >
            <span>
              Dataset overview
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      class="dataset-item border border-solid border-gray-200 leading-5"
    >
      <div
        class="opt-image--container"
      >
        <div
          class="thumb"
        >
          <img
            alt="Optical image"
            class="thumb--img"
            src="allDatasets.2.thumbnailOpticalImageUrl"
            style="transform: scaleY(1) translateY(0%); height: 100px;"
          />
           
          <img
            alt="Ion image thumbnail"
            class="thumb--img thumb--img__ion"
            src="allDatasets.2.ionThumbnailUrl"
            style="transform: scaleY(1) translateY(0%); height: 100px;"
          />
           
          <!---->
        </div>
      </div>
       
      <div
        class="ds-info"
      >
        <div
          class="ds-item-line flex"
        >
          <span
            class="font-bold truncate"
            title="allDatasets.2.name"
          >
            allDatasets.2.name
          </span>
          <mock-el-popover
            class="ml-1"
            placement="top"
            popper-class="min-w-0"
            trigger="hover"
          >
            <div
              slot-key="default"
            >
               
              <p
                class="leading-none m-0"
              >
                Copy dataset id to clipboard
              </p>
            </div>
            <div
              slot-key="reference"
            >
              <button
                class="button-reset flex  w-6 h-6"
              >
                <svg
                  class="h-full w-full text-gray-400 hover:text-gray-500 sm-copy-icon"
                  data-icon="copy-id"
                />
              </button>
            </div>
          </mock-el-popover>
        </div>
        <div
          class="ds-item-line text-gray-700"
        >
          <span
            class="ds-add-filter"
            title="Filter by species"
          >
            allDatasets.2.organism
          </span>
          , 
          <span
            class="ds-add-filter"
            title="Filter by organism part"
          >
            alldatasets.2.organismpart
          </span>
           
          <span
            class="ds-add-filter"
            title="Filter by condition"
          >
            (alldatasets.2.condition)
          </span>
        </div>
        <div
          class="ds-item-line"
        >
          <span
            class="ds-add-filter"
            title="Filter by ionisation source"
          >
            allDatasets.2.ionisationSource
          </span>
           + 
          <span
            class="ds-add-filter"
            title="Filter by analyzer type"
          >
            allDatasets.2.analyzer.type
          </span>
          , 
          <span
            class="ds-add-filter"
            title="Filter by polarity"
          >
            positive mode
          </span>
          , RP 123k @ 1234
        </div>
        <div
          class="ds-item-line"
        >
          Submitted 
          <mock-elapsed-time
            date="allDatasets.2.uploadDT"
          />
           by 
          <span
            class="ds-add-filter"
            title="Filter by submitter"
          >
            allDatasets.2.submitter.name
          </span>
          <span>
            , 
            <mock-el-dropdown
              placement="bottom"
              show-timeout="50"
              trigger="hover"
            >
              <div
                slot-key="default"
              >
                <span
                  class="text-base text-primary cursor-pointer"
                >
                  allDatasets.2.group.shortName
                </span>
              </div>
              <div
                slot-key="dropdown"
              >
                <mock-el-dropdown-menu>
                  <mock-el-dropdown-item
                    command="filter_group"
                  >
                    Filter by this group
                  </mock-el-dropdown-item>
                  <mock-el-dropdown-item
                    command="view_group"
                  >
                    View group
                  </mock-el-dropdown-item>
                </mock-el-dropdown-menu>
              </div>
            </mock-el-dropdown>
          </span>
        </div>
        <div
          class="ds-item-line"
        >
          <span>
            <a
              class=""
              href="/annotations?db_id=6&ds=FINISHED1"
            >
              20 annotations
            </a>
             @ FDR 10% (HMDB - v2.5)
          </span>
        </div>
      </div>
       
      <div
        class="ds-actions"
      >
        <mock-el-dialog
          customclass="dataset-item-dialog"
          title="Provided metadata"
        >
          <div
            class="el-row"
          >
            <mock-el-tree
              data="[object Object],[object Object],[object Object]"
              default-expanded-keys="MS analysis,Detector resolving power,Data Management,Annotation settings"
              id="metadata-tree"
              node-key="id"
            />
          </div>
        </mock-el-dialog>
        <span>
          <i
            class="el-icon-picture"
          />
          <mock-el-popover
            placement="top"
            trigger="hover"
          >
            <div
              slot-key="default"
            >
              <div
                class="db-link-list"
              >
                Select a database:
                <div>
                  <a
                    class=""
                    href="/annotations?db_id=2&ds=FINISHED1"
                  >
                    CHEBI
                  </a>
                </div>
                <div>
                  <a
                    class=""
                    href="/annotations?db_id=6&ds=FINISHED1"
                  >
                    HMDB - v2.5
                  </a>
                </div>
                <div>
                  <a
                    class=""
                    href="/annotations?db_id=22&ds=FINISHED1"
                  >
                    HMDB - v4
                  </a>
                </div>
              </div>
            </div>
            <div
              slot-key="reference"
            >
              <a>
                Browse annotations
              </a>
            </div>
          </mock-el-popover>
          <br />
        </span>
        <div>
          <i
            class="el-icon-view"
          />
          <a
            href="#"
          >
            Show full metadata
          </a>
        </div>
        <div>
          <i
            class="el-icon-data-analysis"
          />
          <a
            class=""
            classname="mr-2"
            href="/dataset/FINISHED1"
          >
            <span>
              Dataset overview
            </span>
          </a>
        </div>
      </div>
    </div>
     
    <!---->
  </div>
   
  <div
    class="mb-8 p-2 flex flex-row justify-end"
  >
    <mock-el-pagination
      class="flex"
      current-page="1"
      hide-on-single-page=""
      layout="prev, pager, next"
      page-size="20"
      pager-count="11"
      total="23"
    />
  </div>
</div>
`;
